### PR TITLE
ledger-api-test-tool: Show a pleasant error message if DAR upload fails.

### PR DESCRIPTION
**Pull request by SamirTalwar-DA**
Previously, we would show a big scary "UNCAUGHT EXCEPTION", which was unnecessary. This is normal on connection failure.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by fabiotudone-da — APPROVED** [review]
If it's not too expensive to write a test for that, please consider doing it.

**Comment by SamirTalwar-DA** [comment]
@fabiotudone-da: Unfortunately we don't have _any_ tests for the Ledger API Test Tool. This is generally OK because we run it against a bunch of ledgers, but there are code paths like this one that will never be triggered in a normal CI run.

I am up for adding unit tests but would rather not do it in this PR, as it's going to be expensive at first. It will require a bunch of Bazel configuration, and a lot of refactoring to extract out the methods under test from the depths of `LedgerTestCasesRunner`.

**Comment by SamirTalwar-DA** [comment]
(BTW, tested manually with _ledger-on-memory_ and _Ctrl+C_.)

**Review by stefanobaghino-da — APPROVED** [review]

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/LedgerTestCasesRunner.scala [hunk 1]
```diff
@@ -208,10 +214,10 @@ final class LedgerTestCasesRunner(
             )(materializer, materializer.executionContext)
           } yield concurrentTestResults ++ sequentialTestResults
 
-        testResults
-          .recover { case NonFatal(e) =>
+        testResults.recover {
```
Comment (miklos-da) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/LedgerTestCasesRunner.scala:217]: You may avoid having to resort to `isInstanceOf` by pattern matching on and rethrowing the `Errors.FrameworkException` first.

Comment (SamirTalwar-DA) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/LedgerTestCasesRunner.scala:217]: Yeah, I tried that but it seemed uglier to me. I don't really mind either way though.

**Pull request merged**